### PR TITLE
editoast: move `Direction` to `editoast_schemas`

### PR DESCRIPTION
**Pull request by hamz2a**
Part of https://github.com/OpenRailAssociation/osrd/issues/7048

### CODE REVIEW

File: editoast/editoast_schemas/src/infra/direction.rs
Comment (woshilapin) [inline editoast/editoast_schemas/src/infra/direction.rs]: I would say this should stay defined next to `TravelDir`, not in the schema since `TravelDir` has no existence in the RailJSON, right? I'd be happy to have another opinion on this (cc @leovalais).

**Review by woshilapin — APPROVED** [review]
Thanks for this PR. Only one question which I'd wait for a response before merging this.

File: editoast/editoast_schemas/src/infra/direction.rs
Comment (hamz2a) [reply editoast/editoast_schemas/src/infra/direction.rs]: I need `Direction` in `editoast_common` for this `From` (Clippy doesn't accept `Into`) in this case.
`editoast_schemas` already depends on `editoast_common`.

Comment (leovalais) [reply editoast/editoast_schemas/src/infra/direction.rs]: I'd say the problem here is that because of its name, `TravelDir` seems to be a railjson object when it's not. It's used to shift a range map (cf. rangemap_utils.rs). However `Direction` is indeed a railjson object (whilst generic enough to be placed in `editoast_schemas::primitives` I'd say).

I'd suggest:

1. rename `TravelDir` to `Direction` with variants `StartToEnd` and `EndToStart` (to match the naming of the crate `rangemap`)
2. rename `travel_range_map` to `fn shift_range_map(range_map, shifting_point, direction)`
3. `impl From<Direction> for editoast_common::rangemap_utils::Direction`, that way we're making it clear it's two very different things

Comment (leovalais) [reply editoast/editoast_schemas/src/infra/direction.rs:24]: I'll fully review this PR tomorrow

**Review by woshilapin — APPROVED** [review]

**Pull request merged**
